src/layer.rs

```diff
@@ -117,7 +117,7 @@ impl Layer {
     }
 
     pub fn get_chunk(&self, chunk_pos: UVec2) -> Option<Entity> {
-        self.chunks[morton_index(chunk_pos)]
+        *self.chunks.get(morton_index(chunk_pos))?
```

Comment:
I'm not sure this is a great fix. I need to think about it a bit more. 🤔

Reply:
I ran into this as well while playing around with things. Does it make more sense to return either a Result<Entity, MapTileError> rather than Option<Entity>? This way OutOfBounds can be returned for this particular scenario and NonExistent could be returned in place of a None value.

Reply:
Would this be acceptable?

```rust
    pub fn get_chunk(&self, chunk_pos: ChunkPos) -> Option<Entity> {
        match self.chunks.get(morton_index(chunk_pos)) {
            Some(Some(chunk)) => Some(*chunk),
            _ => None,
        }
    }
```

Reply:
I'm planning on ripping out the morton encoding(I think it slows things down overall), but I think this a viable solution.